### PR TITLE
Ensure configuration up-to-date on each request.

### PR DESCRIPTION
**Pull request by tdilauro**
## Description

Moves the check to reload configuration if it has changed to a Flask `before_request` lifecycle function.

## Motivation and Context

I happened to notice the location of this call when working on something else. It seemed that is should be moved to a more "expected" place in the codebase for a couple of reasons:
- The check was being performed in only a subset of cases. It would probably be triggered on a fairly regular basis, but there might be some weird edge cases in which it could lead to unexpected and hard to reproduce behavior. 
- It was in an unexpected place.

## How Has This Been Tested?

- Added new tests.
- CI tests for associated feature branch pass.

## Checklist

- N/A - I have updated the documentation accordingly.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/palace/manager/api/controller/base.py [hunk 1]
```diff
@@ -108,21 +108,26 @@ def authenticate(self):
...
+        :param library_short_name: Optional key for looking up the library.
+        :return: The Library for the provided key, if found; a default library,
+            if available; or the `LIBRARY_NOT_FOUND` ProblemDetail.
+
+        If the key is present, use it to look up the library. If it is not
+        found, return the `LIBRARY_NOT_FOUND` ProblemDetail. If the key is
+        not present, try to get a default library. If one is not found, return
+        the `LIBRARY_NOT_FOUND` ProblemDetail.
         """
-        self.manager.reload_settings_if_changed()
```
Comment (tdilauro) [inline src/palace/manager/api/controller/base.py:118]: This deletion is the key change. The additions are just adding some type hints and hopefully improving documentation.

File: tests/manager/api/controller/test_base.py [hunk 2]
```diff
@@ -578,6 +578,10 @@ def test_library_for_request(
             assert expect_default == value
             assert expect_default == flask.request.library  # type: ignore
 
+    # TODO: Remove this test. It's just prove that it passes once in the PR.
+    #  The `reload_settings_if_changed` call has been moved from the relative
+    #  obscurity of `BaseCirculationManagerController.library_for_request`
+    #  into a Flask `before_request` function.
```
Comment (tdilauro) [inline tests/manager/api/controller/test_base.py:584]: Plan to remove this test once the PR has been reviewed.

**Review by jonathangreen — APPROVED** [review]
Looks good to me 🚀

File: tests/manager/api/controller/test_base.py (hunk 2 above)
Comment (tdilauro) [reply tests/manager/api/controller/test_base.py]: This test has been removed.
